src/Microsoft.TemplateEngine.Abstractions/IParameterSymbolLocalizationModel.cs

```diff
@@ -1,13 +1,23 @@
-﻿using System.Collections.Generic;
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
 
 namespace Microsoft.TemplateEngine.Abstractions
 {
     public interface IParameterSymbolLocalizationModel
     {
         string Name { get; }
 
-        string Description { get; }
+        /// <summary>
+        /// Gets the localized friendly name of the symbol to be displayed to the user.
+        /// </summary>
+        string? DisplayName { get; }
+
+        string? Description { get; }
 
-        IReadOnlyDictionary<string, string> ChoicesAndDescriptions { get; }
+        IReadOnlyDictionary<string, ParameterChoiceLocalizationModel> Choices { get; }
```

Comment:
is it safe change? Can it make older versions of `TemplateCache` unreadable? (meaning renaming `ChoicesAndDescriptions`)

Reply:
Version -> Hive -> cache.json file will be different, going forward with Discovery this becomes even safer change, since Cache is versioned and thrown away if cache version is out of date.

Reply:
Lately I realized that we use same class for blob storage cache file, so we cannot change it without adding new version for search cache. 
I think we can change it still if we keep serialization same as before. We can also do this change as we work on improvements for search cache - we will need new version anyway.

Reply:
I'm keeping this change since we will be changing search cache file in .NET 6 anyway. Any concerns?

Reply:
Created the tag [TemplateDiscovery-CacheVersion-1-0](https://github.com/dotnet/templating/releases/tag/TemplateDiscovery-CacheVersion-1-0) from which we can create a branch in the future if needed.

How can we prevent creating a new branch like this every time we make a change to cache file format? Hopefully, once we change the nuget search file to a custom format, changing the cache format won't cause this problem. Is that right?

Reply:
I think we should think carefully of new format and allow only additive changes. If we need modifications, we will need versioning approach.